### ISSUE TITLE
texture-view : getCameraIdList returns no camera

### ISSUE THREAD

**Issue by pateldarsh** [googlesamples/android-ndk#509]
File : camera_manager.cpp

void NDKCamera::EnumerateCamera() {

  ACameraIdList* cameraIds = nullptr;
  CALL_MGR(getCameraIdList(cameraMgr_, &cameraIds));

 **// APIs doesn't give any camera details
  cameraIds->numCameras is zero.**

}

Because of this App is not able to run.

Devices tried: Android Oreo Emulator on Laptop, Moto G4 plus with Android 7.0

**Comment by ggfan** [comment]
you are correct: Android O emulator does not report camera, thanks for reporting ( I did not know that )!
API 27 (O MR1 emulator seems to be ok: reporting correctly with Pixel 2 emulated device API 27 ), may you use that one  Android O MR1 emulator?

**Comment by St2mZ** [comment]
It also reports 0 cameras on Moto G5S, Moto C and Xiaomi Mi 5. Java sample works fine on same devices.

**Comment by ggfan** [comment]
Suspecting it is legacy camera implementation on those devices:

> Most likely these are LEGACY-level devices, which work only in Java

Tried on SM-G930T, the two camera are:

- [Back Facing](https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics.html#LENS_FACING): [FULL](https://developer.android.com/reference/android/hardware/camera2/CameraMetadata.html#INFO_SUPPORTED_HARDWARE_LEVEL_FULL) [hardware_level](https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics.html#INFO_SUPPORTED_HARDWARE_LEVEL)
- Front Facing: [LIMITED](https://developer.android.com/reference/android/hardware/camera2/CameraMetadata.html#INFO_SUPPORTED_HARDWARE_LEVEL_LIMITED)  hardware_level
On native side, sample still gets 2 camera, could run, but could not adjust exposure etc.

The information needs to be retrieved from [Java side](https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics.html#INFO_SUPPORTED_HARDWARE_LEVEL). May you use the snippet code there to pull the camera's hardware level on those devices? something like:
```java
        CameraManager camMgr = (CameraManager)getSystemService(Context.CAMERA_SERVICE);
        try {
            String[] cameraIds = camMgr.getCameraIdList();
            if (cameraIds.length != 0 ) {
                for (String id : cameraIds) {
                    CameraCharacteristics characteristics = camMgr.getCameraCharacteristics(id);
                    int deviceLevel = characteristics.get(CameraCharacteristics.INFO_SUPPORTED_HARDWARE_LEVEL);
int facing = characteristics.get(CameraCharacteristics.LENS_FACING);
                    Log.i("CAM-info ==== ", "Camera " + id + " hardware level = " + deviceLevel 
                              + "Facing " + facing);
                    if (deviceLevel == CameraCharacteristics.INFO_SUPPORTED_HARDWARE_LEVEL_LEGACY) {
                        Log.i("CAM-info====", "Camera " + id + " is legacy camera");
                    }
                    
                    Log.i("CAM-info===", "Camera Facing: " + facing);
                }
            }
        } catch (CameraAccessException e) {
            e.printStackTrace();
        }
```

**Comment by St2mZ** [comment]
Thank you, ggfan!

Indeed, both cameras on Moto G5S are legacy ones. I did not test on other devices. So, native interface is unreliable.

**Comment by alexcohn** [comment]
FWIW, working with legacy cameras via camera2 API is unreliable even in Java. One reason is that, if I understand correctly, the device manufacturers never really tried to do it better.

**Comment by pateldarsh** [comment]
@ggfan I have tried Pixel 2 XL ( API 26, 27, 28, P). None of this supports texture-view example. Could you please specify which combination would work ? Thanks you for your reply.

**Comment by ggfan** [comment]
let me find a device and have a look

**Comment by ggfan** [comment]
may you provide more info about your symptoms on the Pixel XL2 (OS version)?  Tried on the Pixel XL 2 test phone, its physical resolution is 1440 x 2880 ( 1:2 ratio ) so it does not find the matching  ratio in the reported mode list, hence pick 480x640 to use.  Is this what you mean?

**Comment by ggfan** [comment]
@pateldarsh : Yeah, on emulator the two camera controls are not supported with webcam + emulator:

- ACAMERA_SENSOR_INFO_SENSITIVITY_RANGE
- ACAMERA_SENSOR_INFO_EXPOSURE_TIME_RANGE

if you choose "emulated" in the "Advanced Settings", it will be ok ( but you only see the blocky contents, no real camera input from host ).

Merged code to not fail if these 2 controls are not supported, so you would get partial functionality when running on emulator.  I did notice rotation is not correct when it is on emulator, that is different issue from this one: emulator functionality is different from the functionality from real device for this camera.
